### DISCUSSION TITLE
how to build 8BIM profile (file) from scratch ?

### DISCUSSION THREAD

**Discussion by misko2k** [ImageMagick/ImageMagick#6024] [Help]
### ImageMagick version

Magick.NET-Q8-Anycpu

### Operating system, version and so on

VS2022, Win10

### Description

Hi, 

I have picture without any clipping path, and i must after analysing this picture put on certain position clipping mask.

i have seen that you can extract 8BIN profile into file and then apply this to any picture.

so my question would be, how can i build this 8BIM profile (file) with my set of clipping points ?

any help are appreciated

**Comment by snibgo** [comment]
With Gimp, you can create a path interactively, or import it from an SVG file. Then export to a TIFF file. That TIFF file will contain an 8BIM profile that contains the path.

Does that help?

**Reply by misko2k** [reply]
Gimp is not solution for me , i must get it only with code. Now i have decoded all , but i am strugling on coordinates, i dont get them so decoded like on Adobe page described:

`All points used in defining a path are stored in eight bytes as a pair of 32-bit components, vertical component first.
The two components are signed, fixed point numbers with 8 bits before the binary point and 24 bits after the binary point. Three guard bits are reserved in the points to eliminate most concerns over arithmetic overflow. Hence, the range for each component is 0xF0000000 to 0x0FFFFFFF representing a range of -16 to 16. The lower bound is included, but not the upper bound.
This limited range is used because the points are expressed relative to the image size. The vertical component is given with respect to the image height, and the horizontal component is given with respect to the image width. [ 0,0 ] represents the top-left corner of the image; [ 1,1 ] ([ 0x01000000,0x01000000 ]) represents the bottom-right.
In Windows, the byte order of the path point components are reversed; you should swap the bytes when accessing each 32-bit value`

But i get only garbage as value back instead of expected value.

have any idea how to get it converted in C#  ?
